### PR TITLE
Add support for AzureOpenAI

### PR DESCRIPTION
**Pull request by ajac-zero**
**Objetive**

Added a new TextVectorizer for the AzureOpenAI API, which is based on the AzureOpenAI and AsyncAzureOpenAI classes from openai>=1.0.0.

**Reason**

Compatibility with Azure OpenAI is an important feature for developers that are building enterprise AI applications with Azure Cloud. Particularly for use cases where privacy is a concern, and data must stay within their cloud tenant.

My team particularly wants to integrate Semantic Cache (with Azure Cache for Redis Enterprise). Since we rely on the Azure OpenAI API, I created this new vectorizer to add that functionality.

### CODE REVIEW

**Comment by tylerhutcherson** [comment]
@ajac-zero thanks for opening this -- will take a look and we will definitely get this in soon.

**Review by tylerhutcherson — CHANGES_REQUESTED** [review]
Thanks for this contribution! Will you please add tests for this (there are tests for other vectorizers you can follow along) and also a small addition to the docs `user_guide` for vectorizers. Thanks!

**Comment by ajac-zero** [comment]
Sure! I've added the Azure vectorizer to the vectorizer test loop and a new section for Azure OpenAI in the vectorizers notebook.

All vectorizer tests pass with the Azure OpenAI vectorizer.
image

File: redisvl/utils/vectorize/text/azureopenai.py [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+    allowing for batch processing of texts and flexibility in handling
+    preprocessing tasks.
+
+    .. code-block:: python
+
+        # Synchronous embedding of a single text
+        vectorizer = AzureOpenAITextVectorizer(
+            model="text-embedding-ada-002",
+            api_config={
+                "api_key": "your_api_key", # OR set OPENAI_API_KEY in your env
```
Comment (tylerhutcherson) [inline redisvl/utils/vectorize/text/azureopenai.py]:
```suggestion
                "api_key": "your_api_key", # OR set AZURE_OPENAI_API_KEY in your env
```

**Review by tylerhutcherson — CHANGES_REQUESTED** [review]
So close! Left one small typo comment, then I will merge. I will fast follow with a commit to add the env vars to our github actions scripts.

**Review by tylerhutcherson — APPROVED** [review]

**Pull request merged**